packages/openai/src/RPA/OpenAI.py

```diff
@@ -0,0 +1,241 @@
+import openai
+import logging
+from typing import Optional 
+from robot.api.deco import keyword
+
+class OpenAI:
+    """Library to support `OpenAI <https://openai.com>`_ service.
+
+    **Robot Framework example usage**
+
+    .. code-block:: robotframework
+
+        *** Settings ***
+        Library    RPA.Robocorp.Vault
+        Library    OpenAI.py 
+
+        *** Tasks ***
+        Create a text completion
+            ${secrets}   Get Secret   secret_name=OpenAI
+            Authorize To OpenAI   api_key=${secrets}[key] 
+            ${completion}    Completion Create    
+            ...     Write a tagline for an ice cream shop
+            ...     temperature=0.6
+            Log   ${completion}
+    
+    **Python example usage**
+
+    .. code-block:: python
+
+        from RPA.Robocorp.Vault import Vault
+        from OpenAI import OpenAI
+
+        secrets = Vault().get_secret("OpenAI")
+        baselib = OpenAI()
+        baselib.authorize_to_openai(secrets["key"])
+
+        result = baselib.completion_create(
+            Create a tagline for icecream shop',
+            temperature=0.6,
+        )
+        print(result)
+    """
+    
+    ROBOT_LIBRARY_SCOPE = "Global"
+    ROBOT_LIBRARY_DOC_FORMAT = "REST"
+
+    def __init__(self) -> None:
+        self.logger = logging.getLogger(__name__)
+
+    @keyword
+    def authorize_to_openai(self, api_key: str) -> None:
+        """Keyword for authorize to OpenAI with your API key obtained from your account.
+
+        :param api_key: Your OpenAI API key
+
+        Robot Framework example:
+
+        .. code-block:: robotframework
+
+            ${secrets}   Get Secret   secret_name=OpenAI
+            Authorize To OpenAI   api_key=${secrets}[key] 
+
+        Python example:
+
+        .. code-block:: python
+
+            secrets = Vault().get_secret("OpenAI")
+            baselib = OpenAI()
+            baselib.authorize_to_openai(secrets["key"])
+
+        """
+        openai.api_key = api_key
+
+    @keyword
+    def completion_create(self, 
+        prompt: str,
+        model: Optional[str] = "text-davinci-003",
+        temperature: Optional[int] = 0.7,
+        max_tokens: Optional[int] = 256,  
+        top_p: Optional[int] = 1,   
```

Comment:
Pretty bad parameter name.

The underlying OpenAI can use whatever bad naming it can, but we should not have our keyword users suffer the same way.